frontend/src/components/object/ObjectVersion.vue

```diff
@@ -0,0 +1,174 @@
+        Versions
+      </h2>
+    </div>
+    <div class="col-12">
+      <DataTable
+        :value="tableData"
+        data-key="id"
+        class="p-datatable-sm"
+        striped-rows
+        responsive-layout="scroll"
+        :paginator="true"
+        :rows="5"
+        paginator-template="RowsPerPageDropdown CurrentPageReport PrevPageLink NextPageLink "
+        current-page-report-template="{first}-{last} of {totalRecords}"
+        :rows-per-page-options="[5, 10, 20]"
+      >
+        <template #empty>
+          <div
+            v-if="!useAppStore().getIsLoading"
+            class="flex justify-content-center"
+          >
+            <h3>
+              There are no versions associated with this object.
+            </h3>
+          </div>
+        </template>
+        <Column
+          field="updatedAt"
+          header="Version"
+          header-style="width: 33%"
+        >
+          <template #body="{ data }">
+            <div>
+              <router-link
+                v-if="data.id !== props.versionId"
+                :to="{ name: RouteNames.DETAIL_OBJECTS,
+                       query: { objectId: props.objectId, versionId: data.id } }"
+              >
+                {{ formatDateLong(data.updatedAt) }}
+              </router-link>
+              <span v-else>
+                {{ formatDateLong(data.updatedAt) }}
+              </span>
+            </div>
+          </template>
+        </Column>
+        <Column
+          field="createdBy"
+          header="Updated by"
+          header-style="width: 33%"
+        >
+          <template #body="{ data }">
+            <div>
+              {{ data.createdByName }}
+            </div>
+          </template>
+        </Column>
+        <Column
+          header="Actions"
+          header-style="width: 34%"
+          header-class="header-right"
+          body-class="content-right action-buttons"
+        >
+          <template #body="{ data }">
+            <DownloadObjectButton
+              v-if="data.public || permissionStore.isObjectActionAllowed(
+                data.id, getUserId, Permissions.READ, props.bucketId as string)"
+              :mode="ButtonMode.ICON"
+              :ids="['TODO']"
+            />
+            <Button
+              v-if="data.public || permissionStore.isObjectActionAllowed(
+                data.id, getUserId, Permissions.READ, props.bucketId as string)"
+              class="p-button-lg p-button-rounded p-button-text"
+              @click="showInfo('TODO')"
+            >
+              <font-awesome-icon icon="fa-solid fa-circle-info" />
+            </Button>
+            <DeleteObjectButton
+              v-if="permissionStore.isObjectActionAllowed(
+                data.id, getUserId, Permissions.DELETE, props.bucketId as string)"
+              :mode="ButtonMode.ICON"
+              :ids="['TODO']"
+              @on-deleted-success="onDeletedSuccess"
+            />
+          </template>
```

Comment:
As these button actions are not yet wired in, we should temporarily hardcode these as disabled so that if this does show up in prod, people aren't going to trip over incomplete functionality.